python/mlx/nn/layers/pooling.py

```diff
@@ -0,0 +1,155 @@
+
+    - If :attr:`mode` is ``'max'``
+        .. math::
+            out(N_i, k, C_j) = \max_{m=0, \ldots, kL - 1}
+                    \text{input}(N_i, \text{stride[0]} \times k + m, C_j)
+
+    - If :attr:`mode` is ``'mean'``
+        .. math::
+            out(N_i, k, C_j) = \frac{1}{kL} \sum_{m=0, \ldots, kL - 1}
+                    \text{input}(N_i, \text{stride[0]} \times k + m, C_j)
+
+    In two-dimensional case, the output value of the layer with input size :math:`(N, H, W, C)`,
+    output :math:`(N, H_{out}, W_{out}, C)` and :attr:`kernel_size` :math:`(kH, kW)` is:
+
+    - If :attr:`mode` is ``'max'``
+        .. math::
+            \begin{aligned}
+                out(N_i, h, w, C_j) ={} & \max_{m=0, \ldots, kH-1} \max_{n=0, \ldots, kW-1} \\
+                                        & \text{input}(N_i, \text{stride[0]} \times h + m,
+                                                    \text{stride[1]} \times w + n, C_j)
+            \end{aligned}
+    - If :attr:`mode` is ``'mean'``
+        .. math::
+            \begin{aligned}
+                out(N_i, h, w, C_j) =\frac{1} {kH \times kW} & \sum_{m=0, \ldots, kH-1} \sum_{n=0, \ldots, kW-1} \\
+                                        & \text{input}(N_i, \text{stride[0]} \times h + m,
+                                                    \text{stride[1]} \times w + n, C_j)
+            \end{aligned}
+
+    If :attr:`padding` is non-zero, then the input is implicitly padded. If :attr:`mode` is ``'max'``, the input is padded with negative infinity on both sides.
+    If :attr:`mode` is ``'mean'``, the input is padded with 0 on both sides. Padding values are included in the calculation.
+
+    The parameters :attr:`kernel_size`, :attr:`stride`, :attr:`padding`, can either be:
+
+        - a single ``int`` -- in which case the same value is used for all feature axes
+        - a ``list`` of ``int`` -- in which case, the first `int` is used for the first feature axis, the second `int` for the second feature axis, etc.
+
+    Args:
+        kernel_size (int or list(int)): The size of the pooling window.
+        stride (int or list(int)): The stride of the window. Default: :attr:`kernel_size`.
+        padding (int or list(int)): The padding amount on both sides of the feature axis. Default: 0.
+        mode (str): Determines whether to perform max pooling or mean pooling. Either ``'max'`` or ``'mean'``. Default: ``'max'``.
+
+    Examples::
+
+        >>> # max pooling of square window of size=2, stride=2
+        >>> m = nn.Pooling(2, stride=2, mode='max')
+        >>> # mean pooling of square window of size=2, stride=2, padding=1
+        >>> m = nn.Pooling(2, stride=2, padding=1, mode='mean')
+        >>> output = m(input)
+
+    """
+
+    def __init__(
+        self,
+        kernel_size: int | List[int],
+        stride: Optional[int | List[int]] = None,
+        padding: int | List[int] = 0,
+        mode: str = "max",
```

Comment:
The type could be a [literal](https://peps.python.org/pep-0586/):
```suggestion
        mode: Literal["max", "mean"] = "max",
```

Reply:
@ligaz Thank you for the suggestion. It is indeed much better. However, we will drop this argument and use ``MaxPooling1d`` and similar aliases.